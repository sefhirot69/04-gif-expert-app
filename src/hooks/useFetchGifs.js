import {useEffect, useState} from "react";
import getGifs from "../helpers/getGifs";

const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {
        getGifs(category).then(images => {

            setTimeout(() => {
                setState({
                    data: images,
                    loading: false
                })
            }, 1000);
        })
    }, [category]);

    return state;
};

export default useFetchGifs;