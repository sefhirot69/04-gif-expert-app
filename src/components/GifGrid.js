import React from 'react';
import PropTypes from 'prop-types';
import useFetchGifs from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
};

function GifGrid({category}) {

    const {data: images, loading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            <p className='animate__animated animate__flash'>{loading && 'Cargando ...'}</p>
            <div className="card-grid">
                {
                    images.map((img) =>
                        (
                            <GifGridItem
                                key={img.id}
                                {...img}
                            />
                        )
                    )
                }

            </div>
        </>

    );
}

export default GifGrid;