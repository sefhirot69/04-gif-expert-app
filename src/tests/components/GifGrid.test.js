import {shallow} from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre el componente GifGrid', () => {

    const category = 'Goku';

    test('debe mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
    })
});