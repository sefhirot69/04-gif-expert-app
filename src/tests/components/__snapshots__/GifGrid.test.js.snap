// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifGrid debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  />
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas sobre el componente GifGrid debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
